debezium-embedded/src/main/java/io/debezium/embedded/AsyncEmbeddedEngine.java

```diff
@@ -0,0 +1,1266 @@
+                if (taskFuture != null) {
+                    taskFuture.get(); // we need to retrieve the results to propagate eventual exceptions
+                }
+                else {
+                    throw new InterruptedException("Time out while waiting for source task to start.");
+                }
+
+                LOGGER.debug("Started task #{} out of {} tasks.", i + 1, nTasks);
+            }
+            catch (Exception e) {
+                LOGGER.debug("Task #{} (out of {} tasks) failed to start. Failed with {}", i + 1, nTasks, e.getMessage());
+                if (LOGGER.isDebugEnabled()) {
+                    e.printStackTrace();
+                }
+
+                // Store only the first error.
+                if (error == null) {
+                    error = e;
+                }
+                continue;
+            }
+            LOGGER.debug("Calling connector callback after task is started.");
+            connectorCallback.ifPresent(DebeziumEngine.ConnectorCallback::taskStarted);
+        }
+
+        // If at least one task failed to start, re-throw exception and abort the start of the connector.
+        if (error != null) {
+            throw error;
+        }
+    }
+
+    /**
+     * Schedules polling of provided tasks and wait until all polling tasks eventually finish.
+     *
+     * @param tasks {@link List<EngineSourceTask>} of source tasks which should poll for the records.
+     */
+    private void runTasksPolling(final List<EngineSourceTask> tasks)
+            throws ExecutionException {
+        LOGGER.debug("Starting tasks polling.");
+        final ExecutorCompletionService<Void> taskCompletionService = new ExecutorCompletionService(taskService);
+        for (EngineSourceTask task : tasks) {
+            final RecordProcessor processor = selectRecordProcessor();
+            processor.initialize(recordService, transformations, new SourceRecordCommitter(task));
+            taskCompletionService.submit(new PollRecords(task, processor, state));
+        }
+
+        for (int i = 0; i < tasks.size(); i++) {
+            try {
+                taskCompletionService.take().get();
+            }
+            catch (InterruptedException e) {
+                LOGGER.debug("Task interrupted while polling.");
+                Thread.currentThread().interrupt();
+            }
+            LOGGER.debug("Task #{} out of {} tasks has stopped polling.", i, tasks.size());
+        }
+    }
+
+    /**
+     * Select and instantiate {@link RecordProcessor} based on the user configuration.
+     *
+     * @return {@link RecordProcessor} instance which will be used for processing the records.
+     */
+    private RecordProcessor selectRecordProcessor() {
+        // If the change consumer is provided, it has precedence over the consumer.
+        if (handler != null && recordConverter == null) {
+            LOGGER.debug("Using {} processor", ParallelSmtBatchProcessor.class.getName());
```

Comment:
I'd like to see `INFO` log level message saying which processor is used. This should be printed only once, not for every task.